### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by dhaval24**
This pull request updates all the dependencies used by AI-Java SDK to latest versions except Google Guava which is updated to 20.0 as the later versions do no support JDK 1.7

#410 is similar and this PR updates to more recent versions of dependencies

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed
- [ ] CHANGELOG.md updated

### CODE REVIEW

**Comment by dhaval24** [comment]
@grlima @littleaj @Dmitry-Matveev should this PR be good to go ? It passes on Jenkins build

File: gradle/common-java.gradle [hunk 1]
```diff
@@ -32,7 +32,7 @@ if (java7JreDir) {
     def requiredJavaBootArchives = ["rt.jar", "jsse.jar"]
     def bootClasspath = ""
     requiredJavaBootArchives.each { a ->
-        def archivePath = new File(java7JreDir, "lib/$a")
+        def archivePath = new File(java7JreDir, "jre/lib/$a")
```
Comment (littleaj) [inline gradle/common-java.gradle]: the JAVA_JRE_7 dir should point to the JRE root. If you have the JDK, then you point it to JDK_HOME/jre. This breaks if you point it to the JRE and not the JDK.

**Comment by dhaval24** [comment]
@littleaj I have addressed the changes

**Comment by dhaval24** [comment]
@Dmitry-Matveev I have updated the OSS use case for this and thus I am merging this. Let me know if you have any specific concerns here.

**Pull request merged**